### ISSUE TITLE
Gradle builds have stopped (or never did) publishing CFLint version number

### ISSUE THREAD

**Issue by TheRealAgentK** [cflint/CFLint#390]
I noticed in HTML reports that the version of CFLint was empty. That version string comes from Findbugs XML which itself comes from the CFLint XML.

XMLOutput.java gets the version from:

`writer.append("<issues version=\"" + Version.getVersion() + "\"")`

And getVersion itself uses:

```
final InputStream is = Version.class
                .getResourceAsStream("/META-INF/maven/com.github.cflint/CFLint/pom.properties");
```

In my recent build of the -all.jar of 1.2.1-SNAPSHOT with Gradle (using the fatjar task) this directory com.github.cflint with the properties file doesn't exist. I can remember that it DID exist at some point a couple of weeks ago when I dealt with the CFLint version and when I was building with Maven.

Building 1.2.1-SNAPSHOT with mvn install actually works fine and it populates the version correctly. Indeed, pom.properties exists here:

```
#Generated by Maven
#Mon Jul 31 13:37:39 NZST 2017
version=1.2.1-SNAPSHOT
groupId=com.github.cflint
artifactId=CFLint
```

So, I can only assume some recent change to the build setup has changed this or that the Gradle build never had this component.

**Comment by TheRealAgentK** [comment]
Also CCing @mpaluchowski :)

**Comment by mpaluchowski** [comment]
I ran a Gradle build on the code at 8e8998b, so just before the #378 changes in `build.gradle`, and there's _no_ folder at `/META-INF/maven/com.github.cflint/` inside the resulting `-all` JAR. So, if this got broken at some point, it must've been before that.

Having said that, usually the way to read the current package's version is from the `META-INF/MANIFEST.MF` file, specifically the [`Implementation-Version` property](https://docs.oracle.com/javase/tutorial/deployment/jar/packageman.html). This property is also missing from the current JAR (also the one built with Maven), but it could be configured and added.

**Comment by TheRealAgentK** [comment]
Is there any chance either of you guys can add /META-INF/maven/com.github.cflint/ into the Gradle build? Or alternatively implement the other solution - the Java code change to pull the version from a different file would be trivial for me to make.

**Comment by mpaluchowski** [comment]
I'll look at it today.

**Comment by TheRealAgentK** [comment]
Awesome, ta!

**Comment by mpaluchowski** [comment]
I just submitted pull request #393 (have to check how to connect PRs to existing issues, for the future, sorry), adding the `Implementation-Version` to the MANIFEST.MF. Now to **read it**, you can call any class in the package like this:

```java
String version = MyClassName.class.getPackage().getImplementationVersion();
```

I would personally use the **main class** for this, so:

```
CFLintMain.class..getPackage().getImplementationVersion();
```

**Comment by mpaluchowski** [comment]
As for the `pom.properties`, adding it is the default behavior of the `maven-archiver` -- see [property `pomPropertiesFile`](https://maven.apache.org/shared/maven-archiver/index.html).

**Comment by ryaneberly** [comment]
Brilliant, I wasn't familiar with getImplementationVersion().  Run with it @mpaluchowski .

As far as the gradle build, the fatJar was added recently, and the version issue wasn't considered.   mvn has been used for all deployments,  and gradle for the CI stuff on travis.  though I would like to commit to a single build system (presumably migrating towards Gradle).

**Comment by TheRealAgentK** [comment]
Cool, working now as expected! Thx all!
